netmap/netmap_contract.go

```diff
@@ -278,78 +251,37 @@ func AddPeerIR(nodeInfo []byte) {
 	})
 }
 
-// AddPeer accepts information about the network map candidate in the NeoFS
-// binary protocol format, identifies the caller and behaves depending on different
-// conditions listed below.
+// AddPeer proposes a node for consideration as a candidate for the next-epoch
+// network map. Information about the node is accepted in NeoFS API binary
+// format. Call transaction MUST be signed by the public key sewn into the
+// parameter (compressed 33-byte array starting from 3rd byte), i.e. by
+// candidate itself. If the signature is correct, the Notary service will submit
+// a request for signature by the NeoFS Alphabet. After collecting a sufficient
+// number of signatures, the node will be added to the list of candidates for
+// the next-epoch network map ('AddPeerSuccess' notification is thrown after
+// that).
 //
-// Contract settings:
-//
-//	(1) notary-enabled
-//	(2) notary-disabled
-//
-// Callers:
-//
-//	(a) candidate himself, if node's public key corresponds to the signer
-//	(b) Alphabet member
-//	(c) others
-//
-// AddPeer case-by-case behavior:
-//
-//	(1a) does nothing
-//	(1b) panics. Notice that AddPeerIR MUST be used for this purpose.
-//	(2a) throws AddPeer notification with the provided BLOB
-//	(2b) accepts Alphabet vote. If the threshold of votes is reached, adds
-//	new element to the candidate set, and throws AddPeerSuccess notification.
-//	(c) panics
-//
-// Candidate MUST call AddPeer with "online" state in its descriptor. Alphabet
-// members MUST NOT call AddPeer with any other states.
+// Note that if the Alphabet needs to complete information about the candidate,
+// it will be added with AddPeerIR.
 func AddPeer(nodeInfo []byte) {
 	ctx := storage.GetContext()
-	notaryDisabled := storage.Get(ctx, notaryDisabledKey).(bool)
-
-	var ( // for invocation collection without notary
-		alphabet []interop.PublicKey
-		nodeKey  []byte
-	)
-
-	if notaryDisabled {
-		alphabet = common.AlphabetNodes()
-		nodeKey = common.InnerRingInvoker(alphabet)
-	}
 
-	// V2 format
 	publicKey := nodeInfo[nodeKeyOffset:nodeKeyEndOffset]
 
-	// If notary is enabled or caller is not an alphabet node,
-	// just emit the notification for alphabet.
-	if !notaryDisabled || len(nodeKey) == 0 {
-		common.CheckWitness(publicKey)
-		if notaryDisabled {
-			runtime.Notify("AddPeer", nodeInfo)
-		}
-		return
-	}
-
-	candidate := Node{
-		BLOB:  nodeInfo,
-		State: NodeStateOnline,
-	}
-
-	if notaryDisabled {
-		threshold := len(alphabet)*2/3 + 1
-		rawCandidate := std.Serialize(candidate)
-		id := crypto.Sha256(rawCandidate)
-
-		n := common.Vote(ctx, id, nodeKey)
-		if n < threshold {
-			return
-		}
-
-		common.RemoveVotes(ctx, id)
+	common.CheckWitness(publicKey)
+
+	// TODO: is it good approach? We could always call AddPeerIR by the Alphabet,
```

Comment:
This method shouldn't be called in notary-enabled env anyway (except for storage nodes, but it's a no-op then?), so maybe it can be stubbed out.

Reply:
The method requires transaction to be signed by the candidate itself (see `common.CheckWitness` above). So, we can completely abandon the method only by checking on the Inne Ring side. What do you think?

Reply:
@roman-khimov can we remove this condition?

Reply:
No, it'll break things, we have to keep it so that the SN could successfully test-run the method and IR could successfully see the event from a test run of its own, but then ignore it. I will just link nspcc-dev/neofs-node#2278 here.